### ISSUE TITLE
fix flaky test in Location module: LocationTest.shouldGenerateNearbyGPSCoordinateWithOriginInMiles

### ISSUE THREAD

**Issue by cieslarmichal** [cieslarmichal/faker-cxx#999]
LocationTest.shouldGenerateNearbyGPSCoordinateWithOriginInMiles fails some times, debug and fix

**Comment by blakelymayhall** [comment]
Appears to be caused by using the Haversine algorithm to compare geodetic points. https://github.com/cieslarmichal/faker-cxx/blob/main/tests/modules/location_test.cpp#L609. This algorithm assumes the Earth is a sphere and can result in up to 0.5% error when computing geodetic distance. 

I ran this test until it failed and output the lat/lon pairs. Found that using Haversine method, the computed distance was 10.0269 miles, exceeding the 10 mile check.

The fix would be to replace the Haversine algorithm with the more accurate Vincenty method. Using Vincenty method, the computed distance would be 9.9875 miles for the same lat/lon pairs and the test would pass.

Conceivably, this would affect LocationTest.shouldGenerateNearbyGPSCoordinateWithOriginInKilometers as well. 

I'd like to do it as my first issue?
